### PR TITLE
MNT Replace deprcated function findDOMNode

### PR DESCRIPTION
**Pull request by sabina-talipova**
## Parent issue
- https://github.com/silverstripeltd/product-issues/issues/775

### CODE REVIEW

File: client/src/components/Search/Search.js [hunk 1]
```diff
@@ -183,12 +182,12 @@ class Search extends Component {
   /**
    * Try to find the first form field in the advanced form and focus on it.
    */
-  focusFirstFormField(filter = 'input, textarea, select, button') {
+  focusFirstFormField(filter = 'input:not([type=hidden]), textarea, select, button') {
```
Comment (ChloeSartorelli) [inline client/src/components/Search/Search.js]: Why was this change made? It doesn't seem related to the main issue?

File: client/src/components/Search/Search.js [hunk 2]
```diff
@@ -444,40 +444,42 @@ class Search extends Component {
...
-          tagData={this.formatTagData()}
-        >
-
-          <SearchForm
-            id={formId}
-            identifier={identifier}
-            visible={visible}
-            expanded={expanded}
-            formSchemaUrl={formSchemaUrl}
+      <div ref={node => { this.nodeRef = node; }}>
```
Comment (ChloeSartorelli) [inline client/src/components/Search/Search.js]: Don't add a new element to the component - this might mess with css customisations.
Instead, use [forwarding refs](https://legacy.reactjs.org/docs/forwarding-refs.html) and forward a ref from this component down to the `FocusedZone` component.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: client/src/components/Search/Search.js (hunk 1 above)
Comment (sabina-talipova) [reply client/src/components/Search/Search.js]: First field in "Search options" block is hidden. Because of this, when we open "Search options" block the first field is not in focus.

Comment (ChloeSartorelli) [reply client/src/components/Search/Search.js]: Is that somehow related to the issue this PR is for? Or is it separate? If it's separate please handle it in a separate issue and PR to make it clearer what's going on.

File: client/src/components/Search/Search.js (hunk 2 above)
Comment (sabina-talipova) [reply client/src/components/Search/Search.js]: Updated.

File: client/src/components/Search/Search.js (hunk 1 above)
Comment (sabina-talipova) [reply client/src/components/Search/Search.js]: It doesn't work and you cannot test focus on first field in "Search options". I can extract it to another issue if so would be better.

Comment (sabina-talipova) [reply client/src/components/Search/Search.js]: Updated.
PR https://github.com/silverstripe/silverstripe-admin/pull/1567

**Review by GuySartorelli — APPROVED** [review]
Looks good, works well locally. Tested by printing the `node` variable to the console and validating that it's the correct element. This PR doesn't change anything about what _type_ that variable is or what is done with it, so anything beyond that is out of scope.
